cognite/client/_api/datapoints.py

```diff
@@ -959,27 +965,189 @@ def retrieve_dataframe(
+                equidistant datetime index from the earliest `start` to the latest `end` (missing values will be NaNs). If these requirements are not met, a ValueError is raised. Default: False
+            include_aggregate_name (bool): Include 'aggregate' in the column name, e.g. `my-ts|average`. Ignored for raw time series. Default: True
+            include_granularity_name (bool): Include 'granularity' in the column name, e.g. `my-ts|12h`. Added after 'aggregate' when present. Ignored for raw time series. Default: False
+            column_names ("id" | "external_id"): Use either ids or external ids as column names. Time series missing external id will use id as backup. Default: "external_id"
+
+        Returns:
+            pandas.DataFrame: A pandas DataFrame containing the requested time series with a DateTimeIndex localized
+            to the time zone from the start time.
+
+        Examples:
+
+            Get a pandas dataframe in the local timezone in Oslo, Norway:
+
+                >>> from cognite.client import CogniteClient
+                >>> try:
+                ...     from zoneinfo import ZoneInfo
+                ... except ModuleNotFoundError:
+                ...     from backports.zoneinfo import ZoneInfo
+                >>> client = CogniteClient()
+                >>> df = client.time_series.data.retrieve_dataframe_in_tz(
+                ...     id=12345,
+                ...     start=datetime(2023, 1, 1, tzinfo=ZoneInfo("Europe/Oslo")),
+                ...     end=datetime(2023, 2, 1, tzinfo=ZoneInfo("Europe/Oslo")),
+                ...     aggregates="average",
+                ...     granularity="1week",
+                ...     column_names="id")
+
+            Get a pandas dataframe with the sum of the time series with external id "foo" and "bar",
+            for each quarter from 2020 to 2022 returned in the local time zone Oslo, Norway.
+
+                >>> from cognite.client import CogniteClient
+                >>> from datetime import datetime
+                >>> try:
+                ...     from zoneinfo import ZoneInfo
+                ... except ModuleNotFoundError:
+                ...     from backports.zoneinfo import ZoneInfo
+                >>> client = CogniteClient()
+                >>> df = client.time_series.data.retrieve_dataframe(
+                ...     external_id=["foo", "bar"],
+                ...     aggregates=["sum"],
+                ...     granularity="1quarter",
+                ...     start=datetime(2020, 1, 1, tzinfo=ZoneInfo("Europe/Oslo")),
+                ...     end=datetime(2022, 12, 31, tzinfo=ZoneInfo("Europe/Oslo")),
+                ...     )
+        """
+        if (id is not None and external_id is not None) or (id is None and external_id is None):
+            raise ValueError("Either input ids or external ids")
```

Comment:
We have a utility function `exactly_one_is_not_none`:
```suggestion
        if not exactly_one_is_not_none(id, external_id):
            raise ValueError("Either input id(s) or external_id(s)")
```

Reply:
Of course, that makes sense.